pkg/virt-controller/watch/node.go

```diff
@@ -33,28 +33,28 @@ const (
 
 // NodeController is the main NodeController struct.
 type NodeController struct {
-	clientset        kubecli.KubevirtClient
-	Queue            workqueue.RateLimitingInterface
-	nodeInformer     cache.SharedIndexInformer
-	vmiInformer      cache.SharedIndexInformer
-	recorder         record.EventRecorder
-	heartBeatTimeout time.Duration
-	recheckInterval  time.Duration
+	clientset          kubecli.KubevirtClient
+	Queue              workqueue.RateLimitingInterface
+	shadowNodeInformer cache.SharedIndexInformer
```

Comment:
So both the node controller and shadow node controller are watching shadow node changes?  Does it worth to merge them?

Reply:
I would leave this out of the PR but it is likely we could merge the logic of these 2.